tests/MongoDB.Driver.Tests/Linq/MongoQueryableTests.cs

```diff
@@ -1198,6 +1322,24 @@ public void SelectMany_with_collection_selector_syntax_computed_scalar()
                 "{ $project: { __fld0: { $add: ['$C.E.F', '$G.E.F', '$G.E.H'] }, _id: 0 } }");
         }
 
+        [Fact]
+        public void SelectMany_with_collection_selector_syntax_computed_scalar_which_is_called_from_SelectMany()
+        {
+            var selectMany1 = from x in CreateQuery()
+                from g in x.G
+                select g;
+            var selectMany2 = from g in selectMany1
+                from s in g.S
+                select (int?)(g.E.F + s.E.F + s.E.H);
```

Comment:
(int?) is required here just because of these fields x.E.F, 'c.E.F', 'c.E.H' have null values in contents and cannot be converted to int.